### DISCUSSION TITLE
How to communicate variables/memory between workers?

### DISCUSSION THREAD

**Discussion by cayolblake** [walkor/Workerman#726] [Q&A]
Hi @walkor 

I'm implementing a `workerman-memcached` project, yet, want to know if its possible to communicate data across workerman workers.

**Comment by walkor** [comment]
Don't try to communicate between processes. It is very performance consuming.
To avoid inter process communication, Memcache and redis are both single process.

**Reply by cayolblake** [reply]
Why not have something like Swoole's tables?

It uses shared memory/mmap from behind.

Something like https://www.php.net/manual/en/shmop.examples-basic.php ?

**Reply by walkor** [reply]
Swoole is writen in C/C++, which can call C interfaces, such as mmap. However, workerman is developed based on PHP, PHP does not provide such as  mmap interface. Shared memory is a scheme, but it needs to be locked and serialized when reading and writing, which is a very performance consuming operation. Im not sure whether the speed of multi process reading and writing shared memory is higher than that of single process PHP variable.

**Reply by cayolblake** [reply]
Is shmop possible?

Does it make sense to create unix sockets from all threads and communicate between them (dun like it tho) !?

**Reply by walkor** [reply]
https://www.php.net/manual/zh/book.sem.php
https://www.php.net/manual/zh/function.shm-get-var.php
Yes, these are worth trying.

If there are few write operations, using UNIX communication is also a high-performance scheme, but it should be noted that this may cause data inconsistency or delay at a certain time.
